docs/DeveloperGuide.md

```diff
@@ -250,26 +250,123 @@ Priorities: High (must have) - `* * *`, Medium (nice to have)  - `* *`,  Low (un
 Priority | As a ... | I want to ... | So that I can...
 -------- | :-------- | :--------- | :-----------
 `* * *` | new user | see usage instructions | refer to instructions when I forget how to use the App
-`* * *` | user | add a new person |
-`* * *` | user | delete a person | remove entries that I no longer need
-`* * *` | user | find a person by name | locate details of persons without having to go through the entire list
-`* *` | user | hide [private contact details](#private-contact-detail) by default | minimize chance of someone else seeing them by accident
-`*` | user with many persons in the address book | sort persons by name | locate a person easily
-
-{More to be added}
+`* * *` | user | add a new task using natural language | add a task to CMDo
+`* * *` | user | delete a task | remove entries that I no longer need
+`* * *` | user | edit a task| edit the task by accessing the task and typing the changes
+`* * *` | user | find a task by keywords | access details of tasks quickly without having to go through the entire list
+`* * *` | user |only view uncompleted task in CMDo| avoid being confused by completed task that are overdue| 
+`* * *` | user | my tasks sorted by due date and due time|locate urgent tasks easily | 
+`* * *` | user | remove my completed tasks| see only uncompleted tasks | 
+`* * *` | user | book by time slots| block out time for unconfirmed events |
+`* * *` | user | undo my previous action | make mistakes | 
+`* * *` | user | redo my previous action upon undo | make mistakes |
```

Comment:
make mistakes isn't exactly descriptive of why these features are present. Try to add a little more description ~